### PR TITLE
Clean project

### PR DESCRIPTION
**Pull request by BartLucien**
This PR should uniformize the code in the test and hopefully make them run faster. It also "corrects" some code by using the approriate function and move some function in classes where they should be defined instead of in activity directly.

### CODE REVIEW

File: app/src/androidTest/java/com/github/fribourgsdp/radio/AddPlaylistActivityTest.kt [hunk 1]
```diff
@@ -104,7 +120,6 @@ class AddPlaylistActivityTest {
     }
```
Comment (sim0mo) [inline app/src/androidTest/java/com/github/fribourgsdp/radio/AddPlaylistActivityTest.kt:120]: I think you forgot to remove line 119 to release Intents as you do a _@After_ ;)
That might be the reason why tests are failing...

File: app/src/main/java/com/github/fribourgsdp/radio/ImportSpotifyPlaylistsActivity.kt [hunk 2]
```diff
@@ -38,10 +39,12 @@ class ImportSpotifyPlaylistsActivity : AppCompatActivity() {
     companion object {
 
         fun saveInfoToUser(playlistNameToUId: HashMap<String, String>, playlists: Set<Playlist>, ctx : Context) {
-            val user = UserProfileActivity.loadUser(ctx)
-            user.addSpotifyPlaylistUids(playlistNameToUId)
-            user.addPlaylists(playlists)
-            user.save(ctx)
+            User.loadOrDefault(ctx).addOnSuccessListener { u ->
+                u.addSpotifyPlaylistUids(playlistNameToUId)
```
Comment (sim0mo) [inline app/src/main/java/com/github/fribourgsdp/radio/ImportSpotifyPlaylistsActivity.kt:43]: What happens if it fails ? Shouldn't it throw an Exception or something ?

File: app/src/androidTest/java/com/github/fribourgsdp/radio/HostGameHandlerTest.kt [hunk 3]
```diff
@@ -37,7 +38,7 @@ class HostGameHandlerTest {
         handler.handleSnapshot(mockSnapshot)
 
         // Wait for the task of the database to execute
-        Thread.sleep(1000)
+        Thread.sleep(sleepingTime)
```
Comment (Flamantrose321) [inline app/src/androidTest/java/com/github/fribourgsdp/radio/HostGameHandlerTest.kt:41]: I didn't see it in the previous pull requests, but isn't there a way to do this more cleanly without having to put a Thread.sleep?

File: app/src/androidTest/java/com/github/fribourgsdp/radio/HostGameHandlerTest.kt [hunk 4]
```diff
@@ -57,7 +58,7 @@ class HostGameHandlerTest {
...
```
Comment (Flamantrose321) [inline app/src/androidTest/java/com/github/fribourgsdp/radio/HostGameHandlerTest.kt:61]: Same as above

**Review by Flamantrose321 — COMMENTED** [review]
Important update for the maintainability of the project and the cleanliness of the code :)

File: app/src/androidTest/java/com/github/fribourgsdp/radio/HostGameHandlerTest.kt (hunk 3 above)
Comment (BartLucien) [reply app/src/androidTest/java/com/github/fribourgsdp/radio/HostGameHandlerTest.kt:41]: For this you have to see with @VictorSchneuwly but it seemed to be obligatory as a way to let the application have the time to do the things it needs to do. Locally, it works with 1 millisecond as sleepingTime, but it failed once on Cirrus so I increased the value a litlle bit.

**Review by Flamantrose321 — COMMENTED** [review]
I may have the reason why you don't pass the tests.
By putting the mockActivity in the Android test folder, the app is no longer able to find them. As you should see in your Android Manifest. 
Indeed, there are 2 APK generated during project building. First is APK with application and second APK contains test. If you put your activity in test folder it will be in second APK that is used for test and your application APK does not contains it. Maybe that's why your tests don't pass.

Edit by Lucien: this was not the problem. As weird as it might seen, having the manifest buggy doesn't disturb the test.

**Review by Flamantrose321 — APPROVED** [review]
Everything is ok for me

**Review by sim0mo — APPROVED** [review]

**Pull request merged**